### PR TITLE
feedback oriented improvements

### PR DESCRIPTION
**Pull request by christianparpart**
# most important additions:

* ddl2cpp: Drop dead `MakePluralVariableName(table)`
* Adapt `BasicStringBinder<T>::CharType` to work with non-stdlib like strings
* QueryBuilder: Extend `Where(A[, OP], B)` to also allow `B` to be a `SqlQualifiedTableColumnName`
* QueryBuilder: Add intuitive way to test for `NULL`: `Where(A, SqlNullValue)` and `Where(A, std::nullopt)`

### CODE REVIEW

File: src/tests/QueryBuilderTests.cpp [hunk 1]
```diff
@@ -418,6 +418,53 @@ TEST_CASE_METHOD(SqlTestFixture, "SqlQueryBuilder.WhereColumn", "[SqlQueryBuilde
...
+        },
+        QueryExpectations::All(R"(SELECT "foo" FROM "That"
+                                  WHERE "Left1" IS NULL AND "Left2" IS NULL)"));
+
+    checkSqlQueryBuilder(
+        [](SqlQueryBuilder& q) {
+            return q.FromTable("That")
+                .Select()
+                .Field("foo")
+                .Not().Where("Left1", SqlNullValue)
```
Comment (Yaraslaut) [inline src/tests/QueryBuilderTests.cpp:460]: I think this is not the most intuitive API, something like 
```suggestion
                .Where("Left1").Not(SqlNullValue)
```
is better, and can distinguish between complete/incomplete builder state

**Review by Yaraslaut — CHANGES_REQUESTED** [review]
I think it is better to change API slightly

File: src/tests/QueryBuilderTests.cpp (hunk 1 above)
Comment (christianparpart) [reply src/tests/QueryBuilderTests.cpp:460]: I agree. There is also `WhereNot(value)` though. We could add a `WhereNotEquals(a, b)` express inequality.

`Where(a).Not(b)` I fear, might be hard to read too. We should focus on more expressive function member names instead. We already have `WhereNull` and `WhereNotNull` for example.

Comment (Yaraslaut) [reply src/tests/QueryBuilderTests.cpp:460]: so we can hide `Where(a).Not(b)` as implementation detail and provide `WhereNotEquals(a,b)` for the user, what do you think?

Comment (christianparpart) [reply src/tests/QueryBuilderTests.cpp]: @Yaraslaut I've been adding `WhereNotEqual(a, b)` and adapted the tests accordingly. There is no need to add `Where(a).Not(b)` as an implementation detail. (I also cannot think of a way to implement Where(a).Not(b) without breaking current Where(a) semantics).

P.s.: I was generally getting inspired on the query builder API by the Dapper project: https://github.com/DapperLib/Dapper :)

**Review by Yaraslaut — APPROVED** [review]
Thanks
